src/rust/src/http_reader.rs

```diff
@@ -27,8 +27,33 @@ pub struct HttpFgbReader {
 
 impl HttpFgbReader {
     pub async fn open(url: &str) -> Result<HttpFgbReader> {
+        trace!("starting: opening http reader, reading header");
         let mut client = BufferedHttpClient::new(&url);
-        let min_req_size = 512;
+
+        // Because we use a buffered HTTP reader, anything extra we fetch here can
+        // be utilized to skip subsequent fetches.
+        // Immediately following the header is the optional spatial index, we deliberately fetch
+        // a small part of that to skip subsequent requests
+        let prefetch_index_bytes: usize = {
+            // The actual branching factor will be in the header, but since we don't have the header
+            // yet we guess. The consequence of getting this wrong isn't catastrophic, it just means
+            // we may be fetching slightly more than we need or that we make an extra request later.
+            let assumed_branching_factor: usize = 16;
```

Comment:
@pka - you mentioned hoisting some constants to a Config, which I think is a great idea. 

If you'd like I could include that in this PR, or follow up with it later.

Things which might be configurable:

- assumed branching factor

- num_prefetched_layers (not sure if this should be an integer number of layers or a byte count - byte count would probably make more sense to people, but it'd be easy to include 1/2 an index layer, which is unfortunate)

- assumed_header_size?

- combine_request_threshold?

- feature request size

Reply:
I prefer to go with the values you fund and decide later which values should be tunable.

Is `assumed_branching_factor` the same as `PackedRTree::DEFAULT_NODE_SIZE`?

Reply:
Ah yes, I'll reference that constant instead. 

The idea is that it would eventually be configurable, so if I know I'm accessing a FGB with branching factor of 32, I can get enough (and no extra) data.